### PR TITLE
OCPNODE-1257: feat: added unit test for graceful shutdown

### PR DESCRIPTION
**Pull request by eggfoobar**
Added the initial tests for verifying that the kubelet is respecting a pods shutdown grace period when configured for graceful shutdown, [see](https://kubernetes.io/docs/concepts/architecture/nodes/#graceful-node-shutdown)

/cc @jerpeter1

### CODE REVIEW

**Review by jerpeter1 — APPROVED** [review]

**Comment by rphillips** [comment]
lgtm

File: test/extended/pods/graceful-shutdown.go [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+		deleteTestBed(context.Background(), oc)
+	})
+})
+
+func filePathForPod(podName string) string {
+	return fmt.Sprintf("%s/completed-%s", hostPath, podName)
+}
+
+// Wait for the node to come back up after reboot before attempting to validate results
+func waitForNode(oc *exutil.CLI, name string) error {
```
Comment (soltysh) [inline test/extended/pods/graceful-shutdown.go:161]: Any particular reason you can't just use `framework.WaitForNodeToBeReady` from upstream's framework?

**Review by soltysh — APPROVED** [review]
/label tide/merge-method-squash
/approve

File: test/extended/pods/graceful-shutdown.go (hunk 1 above)
Comment (eggfoobar) [reply test/extended/pods/graceful-shutdown.go:161]: Oh I didn't know that existed, didn't see it in the framework package. Created a new PR to use it instead, try and limit duplication https://github.com/openshift/origin/pull/27882 Thanks @soltysh !

**Pull request merged**